### ISSUE TITLE
Port Tokenizer and token processor to dfdx

### ISSUE THREAD

**Issue by swfsql** [swfsql/dfdx-mamba#1]
To reduce the amount of dependencies (or rather concentrate more on dfdx), the Tokenizer (token encoding/decoding) should be ported to dfdx.

This also applies to the token processor, which makes a token selection based on the Vocab output logits.
